test/e2e/nodepool_in_failed_cluster.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+
+	hcpsdk20240610preview "github.com/Azure/ARO-HCP/test/sdk/resourcemanager/redhatopenshifthcp/armredhatopenshifthcp"
+	"github.com/Azure/ARO-HCP/test/util/framework"
+	"github.com/Azure/ARO-HCP/test/util/labels"
+)
+
+var _ = Describe("Customer", func() {
+
+	It("should not be able to deploy nodepool into a hosted cluster with failed provisioning state",
+		labels.RequireNothing,
+		labels.Negative,
+		labels.Critical,
+		func(ctx context.Context) {
+			tc := framework.NewTestContext()
+
+			By("creating resource group ")
+			resourceGroup, err := tc.NewResourceGroup(ctx, "rg-nodepool-in-failed-cluster", tc.Location())
+			Expect(err).NotTo(HaveOccurred())
+
+			clusterName := "nodepool-in-failed-cluster" + rand.String(6)
```

Comment:
Could we name this cluster just "failed-cluster"? It's little confusing that something with nodepool prefix is actually a cluster, and the information about the use case "nodepool-in-failed-cluster" is captured in the resource group already. It will make the errors (if any) easier to read in CI.